File: 2021/7/7.js
"use strict";

let antPos = 1;
let bandLength = 20;

while (antPos < bandLength) {
	const newBandLength = bandLength + 20;
	const stretchRatio = newBandLength / bandLength;
	antPos = antPos * stretchRatio + 1;
	bandLength = newBandLength;
}

console.log(bandLength / 100);

File: 2021/7/7.hs
type AntPos = Double
type BandLength = Integer

step :: (BandLength, AntPos) -> (BandLength, AntPos)
step (bandLength, antPos) = (newBandLength, newAntPos)
    where newBandLength = bandLength + 20
          stretchRatio  = fromIntegral newBandLength / fromIntegral bandLength
          newAntPos = antPos * stretchRatio + 1

main =
    print
    . (`div` 100)
    . fst
    . head
    . dropWhile (\(l, p) -> fromIntegral l > p)
    . iterate step
    $ (20, 1)
